src/mongo/delegates/Views.class.php

```diff
@@ -599,8 +600,37 @@ protected function doJoins($source, $joins, &$dest, $from, $contextAlias, $build
                     {
                         // make sure any sequences are expanded before extracting properties
                         if (isset($ruleset['joins'])) $this->expandSequence($ruleset['joins'],$linkMatch);
-
-                        $dest[_GRAPHS][] = $this->extractProperties($linkMatch,$ruleset,$from);
+                        if (isset($ruleset['filter']))
+                        {
+                            foreach($ruleset['filter'] as $filterPredicate => $filter)
+                            {
+                                foreach($filter as $filterType => $filterMatch)
+                                {
+                                    if (isset($linkMatch[$filterPredicate]))
```

Comment:
Aliases can be added at any time.  Until they're added, a FQ uri would be stored.  You're right about the link match, though.  I could have sworn there was an example where we look up either the alias or the fquri, but I can't find it, so I might be making the whole thing up.